### ISSUE TITLE
About FasterRCNNBoxScoreTarget

### ISSUE THREAD

**Issue by chandlerbing65nm** [jacobgil/pytorch-grad-cam#225]
in the class of `FasterRCNNBoxScoreTarget`, you initialized it with the labels and bounding_boxes that are obtained from function `predict` ---->` targets = [FasterRCNNBoxScoreTarget(labels=labels, bounding_boxes=boxes)]`

Then you calculate the IoU between the initialized boxes and model_outputs["boxes"] which are exactly the same value/coordinates. The output of IoU is always one on this?  For what?

```
class FasterRCNNBoxScoreTarget:
    """ For every original detected bounding box specified in "bounding boxes",
        assign a score on how the current bounding boxes match it,
            1. In IOU
            2. In the classification score.
        If there is not a large enough overlap, or the category changed,
        assign a score of 0.
        The total score is the sum of all the box scores.
    """

    def __init__(self, labels, bounding_boxes, iou_threshold=0.5):
        self.labels = labels
        self.bounding_boxes = bounding_boxes
        self.iou_threshold = iou_threshold

    def __call__(self, model_outputs):
        output = torch.Tensor([0])
        if torch.cuda.is_available():
            output = output.cuda()

        if len(model_outputs["boxes"]) == 0:
            return output

        for box, label in zip(self.bounding_boxes, self.labels):
            box = torch.Tensor(box[None, :])
            if torch.cuda.is_available():
                box = box.cuda()

            ious = torchvision.ops.box_iou(box, model_outputs["boxes"])
            index = ious.argmax()
            if ious[0, index] > self.iou_threshold and model_outputs["labels"][index] == label:
                score = ious[0, index] + model_outputs["scores"][index]
                output = output + score
        return output
```

```
def predict(input_tensor, model, device, detection_threshold):
    outputs = model(input_tensor)
    pred_classes = [coco_names[i] for i in outputs[0]['labels'].cpu().numpy()]
    pred_labels = outputs[0]['labels'].cpu().numpy()
    pred_scores = outputs[0]['scores'].detach().cpu().numpy()
    pred_bboxes = outputs[0]['boxes'].detach().cpu().numpy()
    
    boxes, classes, labels, indices = [], [], [], []
    for index in range(len(pred_scores)):
        if pred_scores[index] >= detection_threshold:
            boxes.append(pred_bboxes[index].astype(np.int32))
            classes.append(pred_classes[index])
            labels.append(pred_labels[index])
            indices.append(index)
    boxes = np.int32(boxes)
    return boxes, classes, labels, indices
```

**Comment by jacobgil** [comment]
Hello,
When we use AblationCAM (and not EigenCAM), I think that's also in the notebook,
The model_output will change every time, since it ablates some of the channels.
Basically if an important channel is ablated the IOU will drop a lot, and then we know the channel is important.

Similarly we could use the category score, or a combination of IOU and the category score, or another property we care about in the bounding boxes.

**Comment by jacobgil** [comment]
I'm closing this issue, please re-open if you have any other questions.
